kroxylicious-filters/kroxylicious-encryption/src/test/java/io/kroxylicious/filter/encryption/InstrumentedKmsTest.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.filter.encryption;
+
+import java.time.Duration;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+import javax.crypto.SecretKey;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import io.kroxylicious.kms.service.DekPair;
+import io.kroxylicious.kms.service.Kms;
+import io.kroxylicious.kms.service.Serde;
+import io.kroxylicious.kms.service.UnknownAliasException;
+import io.kroxylicious.kms.service.UnknownKeyException;
+
+import static io.kroxylicious.filter.encryption.KmsMetrics.OperationOutcome.EXCEPTION;
+import static io.kroxylicious.filter.encryption.KmsMetrics.OperationOutcome.NOT_FOUND;
+import static io.kroxylicious.filter.encryption.KmsMetrics.OperationOutcome.SUCCESS;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class InstrumentedKmsTest {
+
+    @Mock
+    Kms<String, String> kms;
+    @Mock
+    KmsMetrics metrics;
+
+    @Mock
+    SecretKey secretKey;
+
+    @Mock
+    Serde<String> serde;
+
+    @Test
+    void testResolveAliasSuccess() {
+        Kms<String, String> instrument = InstrumentedKms.instrument(kms, metrics);
+        when(kms.resolveAlias("alias")).thenReturn(CompletableFuture.completedFuture("resolved"));
+        CompletionStage<String> stage = instrument.resolveAlias("alias");
+        assertThat(stage).succeedsWithin(Duration.ZERO);
```

Comment:
I think it was just autopilot for me since it's working with futures I wrote the tests assuming something async could be happening. Since we have the backing KMS under control (mocked) we know there's no async work, so could remove it.

But I do think we should be checking that the delegation worked, there should be a check on the value/exception for each test checking we are ferrying things around, I'll add them.